### PR TITLE
ARTS-325 JsonSchema generator tool

### PR DESCRIPTION
**Pull request by anatbal**
## Description
Java code of jsonSchema generator, including a jar file and a README file that explains the use of the tool.

### Changes
- [ART-325] - Build data entities for the json schema- all relevant models are under model folder, perhaps in the future we will want to create a consistency between the trial config models and the models that are necessary today for the terraform generation.

### Checklist:

- [x] Jar that can be writes the jsonSchema file in current directory
- [x] Models that describe the current terraform definition.json file
- [x] README file that explains how to use the tool

### CODE REVIEW

File: build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# json-schema-generator
+
+- This tool generates jsonSchema DRAFT-07 from POJOs using Jackson @Annotations.
+- The jsonSchema is being written to the current directory as "JsonSchema.json".
+- This project uses [mbknor-jackson-jsonschema](https://github.com/mbknor/mbknor-jackson-jsonSchema) package.
+## Using this repository
+
+### Running the jsonSchema tool
+- Run the attached jar: `java -jar json-schema-generator.jar`
+- Alternatively,import the tool as a maven project and run the Main class.
```
Comment (eladiw) [inline build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/README.md:10]:
```suggestion
- Alternatively, import the tool as a maven project and run the Main class.
```

File: build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/SchemaGenerator.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    private final JsonNode jsonSchema;
+
+    public SchemaGenerator(final Class trailConfigFile) {
+        this.jsonSchema = generateJsonSchema(trailConfigFile);
+    }
+
+    private JsonNode generateJsonSchema(final Class trailConfigFile) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonSchemaConfig config = JsonSchemaConfig.vanillaJsonSchemaDraft4()
+                .withJsonSchemaDraft(JsonSchemaDraft.DRAFT_07);
```
Comment (eladiw) [inline build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/SchemaGenerator.java:20]: can you add a comment what is draft 7?

Comment (eladiw) [reply build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/SchemaGenerator.java:20]: if there is a reason

**Review by eladiw — APPROVED** [review]
looks good

File: build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/README.md (hunk 1 above)
Comment (anatbal) [reply build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/README.md]: Done

File: build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/SchemaGenerator.java (hunk 2 above)
Comment (anatbal) [reply build-tools/src/main/java/uk/ac/ox/ndph/mts/json_schema_generator/SchemaGenerator.java:22]: Done

**Pull request closed without merging**